### PR TITLE
Source Harvest: skip FORBIDDEN for EstimateItemCategories stream

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Try to fix this oncall https://github.com/airbytehq/oncall/issues/938

Just skip HTTP 403 FORBIDDEN for `EstimateItemCategories` stream
According to this doc
https://help.getharvest.com/api-v2/estimates-api/estimates/estimate-item-categories/
```
Admin permissions required.
```
I suspect customer has non-admin permissions

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-harvest/source_harvest/streams.py [hunk 1]
```diff
@@ -201,6 +201,13 @@ class EstimateItemCategories(IncrementalHarvestStream):
     Docs: https://help.getharvest.com/api-v2/estimates-api/estimates/estimate-item-categories/
     """
 
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        try:
+            yield from super().read_records(**kwargs)
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code != requests.codes.FORBIDDEN:
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-harvest/source_harvest/streams.py:208]: Should we inform the customer via log message that we are faced with a Forbidden situation?

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-harvest/source_harvest/streams.py [hunk 2]
```diff
@@ -201,6 +201,13 @@ class EstimateItemCategories(IncrementalHarvestStream):
     Docs: https://help.getharvest.com/api-v2/estimates-api/estimates/estimate-item-categories/
     """
 
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-harvest/source_harvest/streams.py:204]: Can other streams face the same issue? I suggest you move this logic to the base class

File: airbyte-integrations/connectors/source-harvest/source_harvest/streams.py [hunk 3]
```diff
@@ -201,6 +201,13 @@ class EstimateItemCategories(IncrementalHarvestStream):
...
+        try:
+            yield from super().read_records(**kwargs)
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code != requests.codes.FORBIDDEN:
+                raise e
+
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-harvest/source_harvest/streams.py:210]:
```suggestion
                raise e
            return []
```

**Pull request merged**